fuel-types/src/array_types.rs

```diff
@@ -367,6 +390,42 @@ impl ContractId {
 
 impl AssetId {
     pub const BASE: AssetId = AssetId::zeroed();
+
+    /// Creates a new AssetId from a ContractId and SubId.
+    ///
+    /// ### Arguments
+    ///
+    /// * `contract_id` - The ContractId of the contract that created the asset.
+    /// * `sub_id` - The SubId of the asset.
+    pub fn new<T: Into<ContractId>>(contract_id: T, sub_id: SubId) -> Self {
+        let contract_id = contract_id.into();
+        let mut hasher = Sha256::new();
+        hasher.update(contract_id);
+        hasher.update(sub_id);
+        let mut bytes = [0u8; 32];
+        bytes.copy_from_slice(&hasher.finalize());
+        Self::from(bytes)
+    }
+
+    /// Creates a new AssetId from a ContractId and the zero SubId. Generally used as the
+    /// default asset of that contract.
+    ///
+    /// ### Arguments
+    ///
+    /// * `contract_id` - The ContractId of the contract that created the asset.
+    pub fn default_for_contract<T: Into<ContractId>>(contract_id: T) -> Self {
+        Self::new(contract_id, SubId::zeroed())
+    }
+
+    /// The base_asset_id represents the base asset of a chain.
+    ///
+    /// ### Additional Information
+    ///
+    /// On the Fuel network, the base asset is Ether. It is hardcoded as the 0x00..00
```

Comment:
It is not true after we made the base asset id configurable. We plan to hide `AssetId::BASE` under `test-helpers` feature in the future, so it is better to remove this method